### DISCUSSION TITLE
When I switch the frontstage toTwo Viewports, an error occurs: "IModel is not open for rpc".

### DISCUSSION THREAD

**Discussion by JingWang111** [iTwin/itwinjs-core#6708] [Q&A]
My code is as follows:

![image](https://github.com/iTwin/itwinjs-core/assets/48142967/0b33406d-c154-41b8-b2e0-a8d20827881a)

![image](https://github.com/iTwin/itwinjs-core/assets/48142967/8280f30a-8481-4c0a-a344-06c94ff93378)

When viewFlag =="syncViewport", I will execute the SyncViewportProps class. However, an error occurs in the code viewState = await iModel?.views.load(defaultViewId); with the message "Uncaught (in promise) Bad Request: IModel is not open for rpc". The defaultViewId is valid and the same as the one used to load the default view.

And I have found another issue: when I comment out the onIModelConnected={_onIModelConnection} in the <Viewer> component, executing the SyncViewportProps class has no problems. What is the reason for this, and how can it be solved?

![image](https://github.com/iTwin/itwinjs-core/assets/48142967/bcb34c4b-e86d-406b-82fa-6fcf4428a565)

**Comment by GerardasB** [comment]
When defining `applicationData` of `sync-viewport-default-content2` make sure to use a `viewState` property, instead of `viewStateClone`, i.e.: `applicationData: { viewState: viewStateClone, ... }`.

**Reply by GerardasB** [reply]
Other than that, based on the original error, the iModel connection is not opened.

**Reply by GerardasB** [reply]
I've added a sandbox to replicate this, but couldn't reproduce the issue (maybe I'm missing something). Feel free to fork it if needed https://www.itwinjs.org/sandboxes/GerardasB/discussion-6708

**Reply by JingWang111** [reply]
Thank you very much for your answer. The problem has been solved. It was caused by my other codes.
